### ISSUE TITLE
[BUG]: Keychron link reciever gets recognised as a controller

### ISSUE THREAD

**Issue by toast003** [PCSX2/pcsx2#12133]
### Describe the Bug

I have a keychron M6, and for some reason it's receiver (the keychron link) shows up as a controller. ![Image](https://github.com/user-attachments/assets/190242f1-bdcb-4eea-90f3-ee4bf27b8dbf)
It's annoying since I sync my pcsx2 config folder with my steam deck, and because the mouse gets detected before the controller I have to redo my mappings every time I switch between the two.

Tried it both with the NixOS package and the flatpak

### Reproduction Steps

Use pcsx2 while having the receiver plugged in

### Expected Behavior

Only the controller gets detected

### PCSX2 Revision

v2.3.99

### Operating System

Linux (64bit) - Specify distro below

### If Linux - Specify Distro

NixOS Unstable

**Comment by toast003** [comment]
Tried it out with the latest nightly on Windows and it also happens, both with sdl and dinput

**Comment by TellowKrinkle** [comment]
> It's annoying since I sync my pcsx2 config folder with my steam deck

PCSX2 is not designed for this to work.  If you're just interested in syncing saves, sync the memcard directory only.  Are there settings in the inis you wish to sync across the two devices?

**Comment by toast003** [comment]
Not really, the only thing that's important to me is the saves. Being able to sync per-game settings would be nice tho

**Comment by NKkrisz** [comment]
I recently bought a Keychron M6 as well and it shows up as a controller in Steam (I don't use PCSX2, just stumbled upon this when searching around).

I think it's the 2.4ghz dongles' firmware that's faulty (both USB-A and USB-C)

Using Fedora Linux 41 (KDE Plasma).

![Image](https://github.com/user-attachments/assets/832ce1e5-8a71-4c11-8444-e0d0abda5954)

**Comment by washurpaws** [comment]
i just disabled the "HID compliant game controller" in device manager under Human Interface Devices (idk how to do this on linux) and my mouse seems to work just fine but dark souls no longer thinks i wanna control it with my mouse. might have broken something that i've yet to notice, but if i haven't noticed is it really a problem?

**Comment by karasuhebi** [comment]
Just found this Issue by googling "Keychron Link." I was also having this issue with the Keychron Link USB receiver that came with my Q6 Max. Just like @washurpaws though disabling the **HID-compliant game controller** (Hardware ID is HID\VID_3434&PID_D030&REV_=300&MI_01&Col02) device fixed it.

This issue was causing the Steam version of Final Fantasy X|X-2 HD Remaster to not take any inputs from my Xbox One S controller.

**Comment by TheSainEyereg** [comment]
In case someone needs it, here are posts on Reddit regardless [Keychron Link issue in general](https://www.reddit.com/r/Keychron/comments/1iphciy/prevent_steam_from_recognizing_keychron_he_board/) and especially [LInux](https://www.reddit.com/r/Keychron/comments/1ltaxbl/keychron_q1_he_detected_as_joystick_on_arch_linux/)

**Comment by TheTechnician27** [comment]
Closing as upstream/external.

* The underlying problem is that the keyboard explicitly misreports what it is.
  * It's an external problem that the manufacturers caused.
* [This is scheduled to be "fixed" in SDL 3.2.26](https://github.com/libsdl-org/SDL/commit/f102ff16a204415b30994d2b77c308eaab30d30d) because SDL will ignore this faulty information as an edge case.
* PCSX2 will likely be updating to SDL 3.2.26 shortly.
  * For now, there's nothing productive that can come from an open issue here.

Edit: I misread the SDL commit; it was in fact released in 3.2.12 and just tagged 3.2.26. So it looks like that still doesn't fix this. Nevertheless, faulty hardware reporting is upstream with SDL at best and external due to the manufacturer's own explicit misreporting at worst. This should be reported there, not here.

**Comment by drandarov-io** [comment]
> i just disabled the "HID compliant game controller" in device manager under Human Interface Devices (idk how to do this on linux) and my mouse seems to work just fine but dark souls no longer thinks i wanna control it with my mouse. might have broken something that i've yet to notice, but if i haven't noticed is it really a problem?

Just wanted to note that this fixes my controller not being detected in multiple steam games like Dark Souls 2 as well. Only commenting so the Google Index might pick this up. It took forever to diagnose because this issue is so niche.

**Comment by themitchnelson** [comment]
> > i just disabled the "HID compliant game controller" in device manager under Human Interface Devices (idk how to do this on linux) and my mouse seems to work just fine but dark souls no longer thinks i wanna control it with my mouse. might have broken something that i've yet to notice, but if i haven't noticed is it really a problem?
> 
> Just wanted to note that this fixes my controller not bein detected in multiple steam games like Dark Souls 2 as well. Only commenting so the Google Index might pick this up. It took forever to diagnose because this issue is so niche.

Thank you! I was having this exact issue with Dark Souls 2: Scholar of the First Sin edition with my Keychron 10 keyboard. I had two "HID compliant game controller" listed in Device Manager and disabled both of them. No downsides so far and I don't have to keep my keyboard unplugged to play the game with the controller.
